src/clr-angular/popover/signpost/signpost-trigger.ts

```diff
@@ -32,26 +33,37 @@ export class ClrSignpostTrigger implements OnDestroy {
   private subscriptions: Subscription[] = [];
   public ariaExpanded: boolean;
   public ariaControl: string;
+  private document: Document;
+  isOpen: boolean;
 
   constructor(
     private ifOpenService: IfOpenService,
-    private renderer: Renderer2,
     private el: ElementRef,
     public commonStrings: ClrCommonStringsService,
     private signpostIdService: SignpostIdService,
+    private signpostFocusManager: SignpostFocusManager,
+    private injector: Injector,
     @Inject(PLATFORM_ID) private platformId: Object
   ) {
+    this.document = this.injector.get(DOCUMENT);
+  }
+
+  ngOnInit() {
+    this.signpostFocusManager.triggerEl = this.el.nativeElement;
     this.subscriptions.push(
       this.ifOpenService.openChange.subscribe((isOpen: boolean) => {
```

Comment:
I think this could be simplified a little.

```js
this.ifOpenService.openChange.subscribe((value: boolean) => {
  this.ariaExpanded = value;
  if (this.isOpen === undefined && typeof value === 'boolean') {
    this.isOpen = value;
    return;
  }

  this.isOpen = value;
  if (!this.isOpen && this.document.activeElement === this.document.body && isPlatformBrowser(this.platformId)) {
    this.singpostFocusManager.focusTrigger();
  }
}
```

* `this.isOpen` and `isOpen` look a lot alike especially in long lines and are hard to read.
*  `else` statement could get skipped.

Reply:
Why are we setting `this.isOpen` to the same thing on each conditional branch? That doesn't make any sense.

Reply:
@bdryanovski I've simplified by using moving most of it into a new method. Also, added some comments explaining that we need to focus only if the signpost goes from a previously open state to closed state. Also, I think made the code reflect that more clearly: 

```
if (!this.isOpen && prevIsOpen) {
 this.focusOnClose();
}
```

@mathisscott we are setting `this.isOpen` to track values emitted by `this.ifOpenService.openChange`. And we need to set the focus only if we detect that the state is changed from open to closed.